spec/notifications-spec.coffee

```diff
@@ -447,3 +421,37 @@ describe "Notifications", ->
             notificationContainer = workspaceElement.querySelector('atom-notifications')
             error = notificationContainer.querySelector('atom-notification.fatal')
             expect(error).toExist()
+
+generateException = ->
+  try
+    a + 1
+  catch e
+    window.onerror.call(window, e.toString(), 'abc', 2, 3, e)
+
+# shortenerResponse
+# packageResponse
+# issuesResponse
+generateFakeAjaxResponses = (options) ->
+  $.ajax.andCallFake (url, settings) ->
+    if url.indexOf('git.io') > -1
+      response = options?.shortenerResponse ? ['--', '201', {getResponseHeader: -> 'http://git.io/cats'}]
+      settings.success.apply(settings, response)
+    else if url.indexOf('atom.io') > -1
+      response = options?.packageResponse ? {
+        repository: url: 'https://github.com/atom/notifications'
+        releases: latest: '0.0.0'
+      }
+      settings.success(response)
+    else
+      response = options?.issuesResponse ? {
+        items: []
+      }
+      settings.success(response)
+
+window.waitsForPromise = (promise) ->
+  promise = promise()
```

Comment:
Man, obviously this is cause you pass `waitsForPromise` a function, not a promise. I need a nap or something.